base/markdown/render/plain.jl

```diff
@@ -37,6 +37,46 @@ function plain(io::IO, list::List)
     end
 end
 
+function plain(io::IO, md::Table)
+    col_widths = reduce(max, map(cell -> map(ansi_length, cell), md.rows))
```

Comment:
Yeah, the idea is for it to give you Markdown back.

Using `ansi_length` doesn't really solve that problem, since it only operates on strings, and not inline markdown elements such as italic. It'd be much easier to render each cell to a string via `plaininline` and then call the usual `length` function.